proxy/src/main/java/net/md_5/bungee/ServerConnector.java

```diff
@@ -329,6 +330,16 @@ public void handle(Login login) throws Exception
 
         bungee.getPluginManager().callEvent( new ServerSwitchEvent( user, from ) );
 
+        if ( !bungee.getConfig().getResourcePack().isEmpty() )
+        {
+            // wait 3 seconds to make sure the client has settled
+            // if we don't wait then the client ignores the request
+            ch.getHandle().eventLoop().schedule( () ->
```

Comment:
Uh there must be a better trigger than this, a packet or something.

Reply:
i have no idea. my tests concluded that waiting 3 seconds is enough, although it may be different on other computers. i also don't like this and im all ears for anybody's suggestions here.

Reply:
tried using DownstreamBridge#connected but nothing happened.

Reply:
Is there a packet the client always sends after changing world?
I guess movement is an option but maybe there is something better

Reply:
Actually why is this sending every switch anyway, shouldn't it only be on the first one?

Reply:
Re packet: will do a research

Re sending every switch: the hash is checked in the method so practically it is being sent only on first join.

Reply:
But why does bungee need to try and override it each time and not just at the start?

Reply:
I wanted to call that code as late as I could without scheduling. Either way if I find such packet you mentioned this is getting removed.

Reply:
after some system.out.println i've concluded that after the join message is sent the server sends a ViewDistance packet:

```
08:30:27 [INFO] downstreambridge: Chat(message={"color":"yellow","translate":"multiplayer.player.joined","with":[{"insertion":"MrIvanPlays","clickEvent":{"action":"suggest_command","value":"/tell MrIvanPlays "},"hoverEvent":{"action":"show_entity","contents":{"type":"minecraft:player","id":"e919bae9-9d06-4dd6-80bf-4947a509e104","name":{"text":"MrIvanPlays"}}},"text":"MrIvanPlays"}]}, position=1, sender=00000000-0000-0000-0000-000000000000)

08:30:27 [INFO] downstreambridge: ViewDistance(distance=11)
```
and never saw a viewdistance packet again. im going to try using it to send the resource pack and see what it turns out.